features/steps/steps.go

```diff
@@ -265,9 +263,32 @@ func (c *Component) privateEndpointsAreNotEnabled() error {
 	return nil
 }
 
-func (c *Component) theDocumentInTheDatabaseShouldBe(id string, doc *godog.DocString) error {
-	// TODO: implement step for verifying documents stored in Mongo. No prior
-	// art of this being done properly in ONS yet so save to be done in future ticket
+func (c *Component) aDocumentInCollectionWithKeyValueShouldMatch(col, key, val string, doc *godog.DocString) error {
+	ctx := context.Background()
+	var expected, result interface{}
+
+	if err := json.Unmarshal([]byte(doc.Content), &expected); err != nil {
+		return errors.Wrap(err, "failed to unmarshal expected document")
+	}
+
+	var bdoc primitive.D
+	if err := c.store.Conn().Collection(col).FindOne(ctx, bson.M{key: val}, &bdoc); err != nil {
+		return errors.Wrap(err, "failed to retrieve document")
+	}
+
+	b, err := bson.MarshalExtJSON(bdoc, true, true)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal bson document")
+	}
+
+	if err := json.Unmarshal(b, &result); err != nil {
+		return errors.Wrap(err, "failed to unmarshal result")
+	}
+
+	if diff := cmp.Diff(result, expected); diff != "" {
+		return fmt.Errorf("-got +expected)\n%s\n", diff)
```

Comment:
This is a super nitpick, but I think traditionally these are the [other way around](https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff):

```go
if diff := cmp.Diff(want, got); diff != "" {
    t.Errorf("MakeGatewayInfo() mismatch (-want +got):\n%s", diff)
}
```